nltk/grammar.py

```diff
@@ -673,8 +674,8 @@ def _calculate_grammar_forms(self):
         prods = self._productions
         self._is_lexical = all(p.is_lexical() for p in prods)
         self._is_nonlexical = all(p.is_nonlexical() for p in prods if len(p) != 1)
-        self._min_len = min(len(p) for p in prods)
-        self._max_len = max(len(p) for p in prods)
+        self._min_len = min(len(p) for p in prods) if len(prods) else None
```

Comment:
This doesn't matter much - but both `min` and `max` support `default` keyword arguments: https://docs.python.org/3/library/functions.html#min
These values are used if the provided iterable is empty.
```suggestion
        self._min_len = min((len(p) for p in prods), default=None)
```
(Note, the extra parentheses are because Python'll complain that generator expressions must be parameterized)
(Btw, if you agree with these changes, you can hit "commit changes" on them)